### ISSUE TITLE
Missing overflow warning on division by -1

### ISSUE THREAD

**Issue by sim642** [goblint/analyzer#1803]
`INT_MIN / -1` is the same as `-INT_MIN`, which is an overflow.

Goblint warns about it if the expression is constant, but is missing a warning when numerator may be `INT_MIN` and denominator may be -1:
```c
#include <limits.h>

int main() {
    int bad = INT_MIN / -1; // WARN
    int x, y;
    if (y != 0) {
        bad = x / y; // TODO WARN
    }
    return 0;
}
```
